store/daystore/day_store.go

```diff
@@ -79,7 +79,8 @@ func NewDayStore(epsilon uint64, resolution time.Duration) *DayStore {
 	// Calculate how many resolutions correspond to an hour
 	hourNS := time.Hour.Nanoseconds()
 	resNS := resolution.Nanoseconds()
-	intervals := uint8(hourNS / resNS)
+	intervals := uint16(hourNS / resNS)
```

Comment:
uint8/16 will be padded if the struct is accompanied by a uint32/64. So if your only reason to use uint8/16 here is space optimization, I would just use uint32.